src/Server.cpp

```diff
@@ -893,6 +918,12 @@ void connectionReceiveBody(Server& server, int clientFd, Connection& connection)
 				connection.m_status = Connection::BuildResponse;
 				server.modifyEvent(clientFd, EPOLLOUT);
 			}
+			else if (connection.m_buffer.size() == Server::s_clientMaxBodySize) {
```

Comment:
Why moved down here? The check should be performed directly after the connection.m_buffer has been updated with the new read data.

Reply:
More closely mimicked connectionReceiveHeader: we track with bytesToRead how many bytes are left to read. 

Before it would always read the full amount of bytes irregardless of how many have already been read. If 1mb is allowed and we already received 900k, we would read another 1mb.
With bytesToRead we check beforehand and only read whats remaining. 

Then we check if we have a complete body. If we don't have complete body, and already read maximum bytes we know that we can never receive the body - all allowed bytes are used. Therefore we can send error

Reply:
Ah yes, I see. Because you added that limit on how many bytes are read, it could be in certain cases that the max. body size has been reached and at the same time the body is complete. If we were to check for the max size before, you would discard a valid request.